api/plan.go

```diff
@@ -203,36 +205,98 @@ func (e FeatureType) minimumSupportedPlan() PlanType {
 	return ENTERPRISE
 }
 
+type featureConfig struct {
+	// whether a particular feature is enabled in the release build. This allows us to progressively develop the feature
+	// without exposing to the public.
+	enabled bool
+	// whether a particular feature is available in [FREE, TEAM, Enterprise] plan respectively.
+	planToggle [3]bool
```

Comment:
nit: consider making a new type for [3]bool.

Reply:
If creating a new type, the type name will still be planToggle. And since it's not exposed to the client and always inline with the planToggle field, there is not much need to create a type. Also having the explicit [3] is helpful for readers to relate with the 3 plans.